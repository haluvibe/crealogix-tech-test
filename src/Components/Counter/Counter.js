import React from "react";
import PropTypes from "prop-types";
import Header from "../Header/Header";
import Button from "../Button/Button";

const Counter = ({ handleIncrementCounter, handleDecrementCounter, count }) => {
  return (
    <>
      <Header title="Counter" />
      <div className="current-count">Current Count: {count}</div>
      <div>
        {count !== 10 && (
          <Button
            handleClick={handleIncrementCounter}
            value="Increment Counter"
          />
        )}
        {count !== 0 && (
          <Button
            handleClick={handleDecrementCounter}
            value="Decrement Counter"
          />
        )}
      </div>
    </>
  );
};

Counter.propTypes = {
  handleIncrementCounter: PropTypes.func.isRequired,
  handleDecrementCounter: PropTypes.func.isRequired,
  count: PropTypes.number.isRequired
};

export default Counter;
