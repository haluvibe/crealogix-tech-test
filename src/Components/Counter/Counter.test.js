import React from "react";
import { shallow } from "enzyme";
import Counter from "./Counter";

const handleIncrementCounter = jest.fn();
const handleDecrementCounter = jest.fn();
const count = 0;

const mockProps = {
  handleIncrementCounter,
  handleDecrementCounter,
  count
};

const setUp = (props = mockProps) => {
  const wrapper = shallow(<Counter {...props} />);
  return wrapper;
};

describe("Counter", () => {
  describe("When count prop equals 0", () => {
    let wrapper;
    beforeEach(() => {
      wrapper = setUp();
    });

    it("should render correctly without errors", () => {
      expect(wrapper).toMatchSnapshot();
    });

    it("includes the Header component", () => {
      expect(wrapper.find("Header").props()).toEqual({
        title: "Counter"
      });
    });

    it("includes a div with text and the counter prop value", () => {
      expect(wrapper.find("div.current-count").text()).toEqual(
        "Current Count: 0"
      );
    });

    it("includes only the increment button", () => {
      expect(wrapper.find("Button").props()).toEqual({
        value: "Increment Counter",
        handleClick: handleIncrementCounter
      });
      expect(wrapper.find("Button").length).toEqual(1);
    });
  });

  describe("When count prop equals 10", () => {
    let wrapper;
    beforeEach(() => {
      mockProps.count = 10;
      wrapper = setUp(mockProps);
    });

    it("includes a div with text and the counter prop value", () => {
      expect(wrapper.find("div.current-count").text()).toEqual(
        "Current Count: 10"
      );
    });

    it("includes only the decrement button", () => {
      expect(wrapper.find("Button").props()).toEqual({
        value: "Decrement Counter",
        handleClick: handleDecrementCounter
      });
      expect(wrapper.find("Button").length).toEqual(1);
    });
  });
});
