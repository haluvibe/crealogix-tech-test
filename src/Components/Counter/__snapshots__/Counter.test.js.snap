// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter When count prop equals 0 should render correctly without errors 1`] = `
<Fragment>
  <Header
    title="Counter"
  />
  <div
    className="current-count"
  >
    Current Count: 
    0
  </div>
  <div>
    <Button
      handleClick={[MockFunction]}
      value="Increment Counter"
    />
  </div>
</Fragment>
`;
